### PR TITLE
[Cherry-pick] Support cleanly resetting driver and hot-plugging sensors.

### PR DESCRIPTION
**Pull request by 130s**
This is a cherry-pick of the changes that was merged in at https://github.com/plusone-robotics/realsense/pull/28 into an older development branch on this fork.

## Branch
- Base: [por_development_202010](https://github.com/plusone-robotics/realsense/compare/por_development_202010), which is a new branch that was checked out from the `tag` [2.2.17](https://github.com/IntelRealSense/realsense-ros/releases/tag/2.2.17).
- Target: `por_development_202010`

## Review items includes:
1. Dev-test. This will likely be done on a in-house, integrated application.
   - [ ] Realsense ROS node starts.
   - [ ] Re-spawning functionality that was added in https://github.com/plusone-robotics/realsense/pull/28 works.
   - Else?
1. Code review
   - [ ] Whether newer changes in the upstream repo's [development](https://github.com/IntelRealSense/realsense-ros/tree/development) is not removed unintentionally.

## TODOs post-merge
1. [ ] Tag on plusone-robotics' fork.
1. [ ] PR to the Intel's upstream.

### CODE REVIEW

**Comment by 130s** [comment]
@malban Would you have time to review today 11/4W if you can spot outstanding logic change?

File: realsense2_camera/src/realsense_node_factory.cpp [hunk 1]
```diff
@@ -244,47 +266,89 @@ void RealSenseNodeFactory::onInit()
...
 						{
+							ROS_DEBUG("Waiting for device...");
+
+ 
 							std::chrono::milliseconds timespan(6000);
 							while (_is_alive && !_device)
 							{
-								getDevice(_ctx.query_devices());
+								ROS_DEBUG("Checking for device...");
+								_device = getDevice();
```
Comment (malban) [inline realsense2_camera/src/realsense_node_factory.cpp]: This call to getDevice() is not in sync with the updated getDevice() method

File: realsense2_camera/src/realsense_node_factory.cpp [hunk 2]
```diff
@@ -244,47 +266,89 @@ void RealSenseNodeFactory::onInit()
...
+ 								// getDevice(_ctx.query_devices());  // This line is obtained from the latest upstream.
 								if (_device)
 								{
-									std::function<void(rs2::event_information&)> change_device_callback_function = [this](rs2::event_information& info){change_device_callback(info);};
-									_ctx.set_devices_changed_callback(change_device_callback_function);
-									StartDevice();
+									ROS_DEBUG("got device");
+									if (_initial_reset)
```
Comment (malban) [inline realsense2_camera/src/realsense_node_factory.cpp]: New new getDevice() method appears to handle the initial reset logic, so I think this section can be removed

File: realsense2_camera/src/base_realsense_node.cpp [hunk 3]
```diff
@@ -888,6 +914,47 @@ void BaseRealSenseNode::setupPublishers()
     }
 }
 
+void BaseRealSenseNode::addMonitoredTopic(const stream_index_pair& stream, Topic topic, const std::string& name)
+{
+    const std::lock_guard<std::mutex> lock(_topic_monitor_mutex);
+    TopicInfo info;
```
Comment (malban) [inline realsense2_camera/src/base_realsense_node.cpp:920]: The TopicInfo definition and topic monitoring method definitions appear to be missing from the header file.

**Comment by 130s** [comment]
I confirmed the claimed functionality still works with the standalone Realsense driver with `rs_camera.launch respawn:=true` option.

Because while making this PR merge conflict inevitably happened, I'll wait for our POR's internal QA process (that integrates Realsense driver into our internal application) for vetting before merging.

**Review by abhijitmajumdar — COMMENTED** [review]
Team decision was made to merge this and test later to expedite the process of release. Have not reviewed this thoroughly, any issues found later should be dealt with in a new issue. Merging.

**Pull request merged**